### ISSUE TITLE
Training epochs show high mAP, but final Validating best.pt line gives near-zero mAP for the same dataset; reproduced on YOLOv11-seg & YOLOv8-seg, Ultralytics 8.3.228

### ISSUE THREAD

**Issue by semihozenc** [ultralytics/ultralytics#22785]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Val

### Bug

I encountered a reproducible issue with Ultralytics YOLOv8-seg and YOLO11-seg models:

During training, the validation metrics (Box & Mask mAP, P, R) are excellent (e.g., Mask mAP@0.5 ≈ 0.97, Mask mAP50-95 ≈ 0.70).

But right after training, Ultralytics runs a final

Validating best.pt
step — and all Mask metrics collapse to near-zero, such as:
	•	Mask mAP@0.5: 0.01
	•	Mask mAP50-95: 0.002
	•	Mask Precision: 0.01
	•	Mask Recall: 0.006

<img width="1417" height="697" alt="Image" src="https://github.com/user-attachments/assets/db74aa4b-66e5-45be-9db0-8c70291d0d8b" />

<img width="1407" height="697" alt="Image" src="https://github.com/user-attachments/assets/a2e4b9d1-78da-47a4-a856-cc382563ab18" />

### Environment

Tested in two independent environments

1. Ultralytics Official Docker

ultralytics/ultralytics:latest
Python 3.11.13
torch 2.8.0+cu128
Ultralytics 8.3.228
NVIDIA RTX 3090
CUDA 12.8

2. Brand new pulled docker instance
Pulled using:

docker run --gpus all -it --ipc=host \
  --name test_env \
  -v /home/datasets:/workspace \
  ultralytics/ultralytics:latest bash

Same issue reproduced.

Dataset
•	Single-class segmentation
•	Standard YOLO polygon TXT format
•	Example label:
0 0.4629 0.3105 0.459 0.3145 0.459 0.3164 ... 0.4727 0.3105

data.yaml:
path: /workspace/yolo_ct_balanced_final
train: train/images
val: val/images
test: test/images
names: [lesion]

### Minimal Reproducible Example

Training commands

yolo train \
  task=segment \
  model=yolo11m-seg.pt \
  data=/workspace/yolo_ct_balanced_final/data.yaml \
  single_cls=true \
  imgsz=640 \
  epochs=50 \
  batch=16 \
  device=0 \
  optimizer=AdamW \
  seed=42 deterministic=True \
  workers=8 \
  rect=False amp=False \
  retina_masks=True overlap_mask=True \
  translate=0.0 scale=0.0 mosaic=0.0 \
  save_period=5 \
  plots=True save_json=True save_txt=True save_conf=True save_crop=True \
  project=/workspace/runs name=deneme2 exist_ok=True \
| tee /workspace/runs/deneme2/train.log

yolo train \
  task=segment \
  model=yolov8m-seg.pt \
  data=/workspace/yolo_ct_balanced_final/data.yaml \
  single_cls=true \
  imgsz=640 \
  epochs=50 \
  batch=16 \
  device=0 \
  optimizer=AdamW \
  seed=42 deterministic=True \
  workers=8 \
  rect=False amp=False \
  retina_masks=True overlap_mask=True \
  translate=0.0 scale=0.0 mosaic=0.0 \
  save_period=5 \
  plots=True save_json=True save_txt=True save_conf=True save_crop=True \
  project=/workspace/runs name=yolov8deneme exist_ok=True \
| tee /workspace/runs/yolov8deneme/train.log

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @semihozenc, thanks for the detailed report and for testing both YOLOv8-seg and YOLO11-seg across multiple environments 🚀 This is an automated triage response — an Ultralytics engineer will review your issue and follow up with you soon.

From your description, this appears to be a 🐛 **Bug Report** related to validation behavior on the final `Validating best.pt` step with segmentation metrics.  

You’ve already provided a strong minimal reproducible example (MRE) with commands and environment details 🙌. If possible, please also include:
- A tiny sample of your dataset (e.g., a few representative images + labels for `train` and `val`)
- The exact `val` command (if you ran a standalone `yolo val` after training)
- Any relevant snippets from the training log around:
  - The last epoch’s validation metrics
  - The final `Validating best.pt` block

These will help the engineering team reproduce and investigate the discrepancy more quickly.

In the meantime, we recommend reviewing the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples that may provide additional context around validation, segmentation, and model saving behavior.

If this turns out to be related to custom training nuances rather than a core bug, please also confirm you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- For real-time chat and sharing logs or screenshots, head to [Discord](https://discord.com/invite/ultralytics) 🎧
- For more structured Q&A, visit [Discourse](https://community.ultralytics.com/)
- For community experiments and discussions, check out our [Subreddit](https://www.reddit.com/r/Ultralytics/)  

## Upgrade

Before the engineering team digs in further, please ensure you have tested with the latest `ultralytics` version (even if you’re already on a recent one), along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python≥3.8** environment with **PyTorch≥1.8**:

```bash
pip install -U ultralytics
```

This helps confirm the behavior still occurs on the current release.

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Since you’re already using the official Ultralytics Docker image, this information will help confirm parity with our verified setups.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Remove these from training command:

```
save_json=True save_txt=True save_conf=True save_crop=True amp=False retina_masks=True
```

**Comment by semihozenc** [comment]
yolo train \
  task=segment \
  model=/workspace/runs/deneme/weights/best.pt \
  data=/workspace/yolo_ct_balanced_final/data.yaml \
  single_cls=true \
  imgsz=640 \
  epochs=5 \
  batch=16 \
  device=0 \
  optimizer=AdamW \
  seed=42 deterministic=True \
  workers=8 \
  rect=False \
  overlap_mask=True \
  translate=0.0 scale=0.0 mosaic=0.0 \
  plots=True \
  project=/workspace/runs name=deneme_5more exist_ok=True \
| tee /workspace/runs/deneme_5more/train.log
Ultralytics 8.3.231 🚀 Python-3.11.14 torch-2.9.1+cu128 CUDA:0 (NVIDIA GeForce RTX 3090, 24036MiB)
engine/trainer: agnostic_nms=False, amp=True, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/workspace/yolo_ct_balanced_final/data.yaml, degrees=0.0, deterministic=True, device=0, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, epochs=5, erasing=0.4, exist_ok=True, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=/workspace/runs/deneme/weights/best.pt, momentum=0.937, mosaic=0.0, multi_scale=False, name=deneme_5more, nbs=64, nms=False, opset=None, optimize=False, optimizer=AdamW, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False, project=/workspace/runs, rect=False, resume=False, retina_masks=False, save=True, save_conf=False, save_crop=False, save_dir=/workspace/runs/deneme_5more, save_frames=False, save_json=False, save_period=-1, save_txt=False, scale=0.0, seed=42, shear=0.0, show=False, show_boxes=True, show_conf=True, show_labels=True, simplify=True, single_cls=True, source=None, split=val, stream_buffer=False, task=segment, time=None, tracker=botsort.yaml, translate=0.0, val=True, verbose=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=3.0, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None
Overriding class names with single class.

                   from  n    params  module                                       arguments
  0                  -1  1      1856  ultralytics.nn.modules.conv.Conv             [3, 64, 3, 2]
  1                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]
  2                  -1  1    111872  ultralytics.nn.modules.block.C3k2            [128, 256, 1, True, 0.25]
  3                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]
  4                  -1  1    444928  ultralytics.nn.modules.block.C3k2            [256, 512, 1, True, 0.25]
  5                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]
  6                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]
  7                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]
  8                  -1  1   1380352  ultralytics.nn.modules.block.C3k2            [512, 512, 1, True]
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]
 10                  -1  1    990976  ultralytics.nn.modules.block.C2PSA           [512, 512, 1]
 11                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 12             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 13                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]
 14                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']
 15             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 16                  -1  1    542720  ultralytics.nn.modules.block.C3k2            [1024, 256, 1, True]
 17                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]
 18            [-1, 13]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 19                  -1  1   1511424  ultralytics.nn.modules.block.C3k2            [768, 512, 1, True]
 20                  -1  1   2360320  ultralytics.nn.modules.conv.Conv             [512, 512, 3, 2]
 21            [-1, 10]  1         0  ultralytics.nn.modules.conv.Concat           [1]
 22                  -1  1   1642496  ultralytics.nn.modules.block.C3k2            [1024, 512, 1, True]
 23        [16, 19, 22]  1   3718003  ultralytics.nn.modules.head.Segment          [1, 32, 256, [256, 512, 512]]
YOLO11m-seg summary: 253 layers, 22,359,987 parameters, 22,359,971 gradients, 113.5 GFLOPs

Transferred 711/711 items from pretrained weights
Freezing layer 'model.23.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
Downloading https://github.com/ultralytics/assets/releases/download/v8.3.0/yolo11n.pt to 'yolo11n.pt': 100% ━━━━━━━━━━━━ 5.4MB 26.5MB/s 0.2s
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.0±0.0 ms, read: 273.7±80.2 MB/s, size: 157.8 KB)
train: Scanning /workspace/yolo_ct_balanced_final/train/labels.cache... 35309 images, 5296 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 35309/35309 841.5Mit/s 0.0s
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 165.2±44.5 MB/s, size: 110.3 KB)
val: Scanning /workspace/yolo_ct_balanced_final/val/labels.cache... 4500 images, 1800 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 4500/4500 60.1Mit/s 0.0s
Plotting labels to /workspace/runs/deneme_5more/labels.jpg...
optimizer: AdamW(lr=0.01, momentum=0.937) with parameter groups 115 weight(decay=0.0), 126 weight(decay=0.0005), 125 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 8 dataloader workers
Logging results to /workspace/runs/deneme_5more
Starting training for 5 epochs...

      Epoch    GPU_mem   box_loss   seg_loss   cls_loss   dfl_loss  Instances       Size
        1/5      9.33G       1.12      1.589     0.6459       1.15         11        640: 100% ━━━━━━━━━━━━ 2207/2207 4.3it/s 8:32
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95)     Mask(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 141/141 3.3it/s 42.1s
                   all       4500       3191          0          0          0          0          0          0          0          0

**Comment by Y-T-G** [comment]
Also remove `optimizer=AdamW`
